helix-core/src/main/java/org/apache/helix/model/ResourceConfig.java

```diff
@@ -775,17 +794,38 @@ private void validate() {
           }
         }
       }
+
+      if (_partitionCapacityMap != null) {
+        if (_partitionCapacityMap.keySet().stream()
+            .noneMatch(partition -> partition.equals(DEFAULT_PARTITION_KEY))) {
+          throw new IllegalArgumentException(
+              "Partition capacity is configured without the DEFAULT capacity!");
+        }
+        if (_partitionCapacityMap.values().stream()
+            .anyMatch(capacity -> capacity.values().stream().anyMatch(value -> value < 0))) {
+          throw new IllegalArgumentException(
+              "Partition capacity is configured with negative capacity value!");
+        }
+      }
     }
 
     public ResourceConfig build() {
       // TODO: Reenable the validation in the future when ResourceConfig is ready.
       // validate();
 
-      return new ResourceConfig(_resourceId, _monitorDisabled, _numPartitions, _stateModelDefRef,
```

Comment:
Builder pattern usually only has one line for the method
```
return new ResourceConfig(this);
```

Reply:
Discussed with Yi offline. I have converted to be the preferred style.
However, the builder pattern used in the resource config is still not the recommended way. We will fix it in the future.